README.md

```diff
@@ -92,9 +92,17 @@ Feel free to enable it and provide feedback about this functionality.
 ## Static pod templates
 
 In addition to that, in the **Kubernetes Pod Template** section, we need to configure the image that will be used to 
-spin up the agent pod. We do not recommend overriding the `jnlp` container except under unusual circumstances. 
-For your agent, you can use the default Jenkins agent image available in [Docker Hub](https://hub.docker.com). In the
-‘Kubernetes Pod Template’ section you need to specify the following (the rest of the configuration is up to you):
+spin up the agent pod. The Jenkins agent runs by default in a container named `jnlp` (historical name, kept for backward compatibility).
+
+You may use it, use a different agent container name and/or provide your own container image.
+
+To avoid drift between the Jenkins controller and your agent, we recommend *not* to include the agent JAR in your image, but instead use the *agent injection* checkbox.
+
+We require a JRE in the provided container image that is compatible with the Jenkins controller version.
+
+For your agent, you can use the default Jenkins agent image available in [Docker Hub](https://hub.docker.com).
```

Comment:
Better to link to the actual image page.

Anyway this sentence sort of contradicts the one two above (_To avoid drift…_).